tools/hsmtool.c

```diff
@@ -772,7 +782,21 @@ int main(int argc, char *argv[])
 		else
 			version = BIP32_VER_MAIN_PRIVATE;
 
-		return dumponchaindescriptors(argv[2], NULL, version);
+		if (argc > 4)
+			show_secrets_str = argv[4];
+
+		if (show_secrets_str) {
+			if (streq(show_secrets_str, "true"))
+				show_secrets = true;
+			else if (streq(show_secrets_str, "false"))
+				show_secrets = false;
+			else
+				errx(ERROR_USAGE, "show_secrets should be "
+						  "either true or false.");
```

Comment:
why do not make it a flag? without true or false? because there there is the possibility of `True` or `TRUE` or any other combination.

Well ofc you can lower it or another trick, maybe also return the error is right, however you can use the idea: If the `--show_secret` is specified it is true, if not it is false

Reply:
The option, like you suggest, is better.

Reply:
I was hesitant with this because so far `hsmtool` is only parsing command line arguments by hand, instead of using any library. But I think I have figured out how to properly parse an option. Please check again 2527122.